### DISCUSSION TITLE
How Typematic is compared vs TSOA?

### DISCUSSION THREAD

**Discussion by jamesxv7** [ts-api-spec/typematic#4] [Q&A] [answered]
Every time I see this type of initiative I get very excited, I think there is a great need to have this kind of system to help produce more maintainable APIs using JS/TS.

My question is how does this proposal compare with TSOA (https://github.com/lukeautry/tsoa)? TSOA is able to generate fully typed API endpoints based on the code generation and controllers, is it Typematic's intention to avoid this step?

**Comment by ecyrbe** [comment, marked as answer]
The Big difference is  Indeed that there is no code generation at all.
It's a base library for other libraries like [Zodios](https://www.zodios.org) to leverage upon.
Zodios already has some of the features of typematic, but is tied to zod schemas.
Typematic will allow client and server libraries using REST to not reinvent the wheel and not need to be a typescript expert to do so.
End users should probably not be aware of the existance of typematic, but use higher level libraries that consume typematic
